feat: Save rule results against collection log

benscobie opened the pull request:
### Description

This PR introduces a meta column to collection logs for storing arbitrary, but typesafe data. In this PR I am saving the rule evaluation results against the added & removed media collection log. These can then be viewed by clicking the meta link next to the log message. Manual additions and removals are also labelled with `(manual)`.

This was cherry-picked from the work in #1705.

### Related issue

This should help in debugging #1446 - and users often wonder why something got added or removed so this should help those guys out too.

### How to test

1. Create a collection with some rules that will match some media
2. Run rules
3. Collection -> Info
4. A meta link should appear next to the log message. Clicking that should bring up a modal similar to test media with the rule results displayed.

ydkmlt84 reviewed (commented):
Functionally, this worked. However the meta information seems a little off, because it is adding an operator at the end. I dont think it does anything, just something I noticed. 

This is from Test Media
```
- plexId: 110403
  result: true
  sectionResults:
    - id: 0
      result: true
      ruleResults:
        - operator: OR
          action: in_last
          firstValueName: Plex - Date added
          firstValue: 2025-05-18T23:22:20.000Z
          secondValueName: custom_days
          secondValue: 2025-05-15T14:45:21.428Z
          result: true
```

and this is from the meta on that same item
```
plexId: 110403
result: true
sectionResults:
  - id: 0
    result: true
    ruleResults:
      - operator: OR
        action: in_last
        firstValueName: Plex - Date added
        firstValue: 2025-05-18T23:22:20.000Z
        secondValueName: custom_days
        secondValue: 2025-05-15T14:43:31.345Z
        result: true
    operator: OR
```

also the `(meta)` link wasn't super obvious that is what I should be seeing/clicking on. maybe an icon/button?

benscobie commented:
@ydkmlt84 I've fixed the operator incorrectly appearing on the first section. Turns out this was an existing bug but didn't show up as test media only runs rules for 1 media item.

I've also replaced the link with a button icon. I've kept the `(manual)` text next to the message as it's just a label.

ydkmlt84 reviewed (approved):
👍

jorenn92 commented:
:tada: This PR is included in version 2.16.0 :tada:

The release is available on [GitHub release](https://github.com/jorenn92/Maintainerr/releases/tag/v2.16.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
